programming_examples/ml/softmax/softmax_whole_array_placed.py

```diff
@@ -0,0 +1,250 @@
+        # in that column and link them to the memory tile FIFOs.
+        inA_fifos_split = [[], [], [], []]
+        for i in range(n_col):
+            inA_fifos_split[i] = inA_fifos[
+                i * n_cores_per_col : (i + 1) * n_cores_per_col
+            ]
+
+        # Link the input FIFOs from memory tiles to the cores in each column
+        for i in range(n_col):
+            object_fifo_link(of_in[i], inA_fifos_split[i], [], of_a_offsets)
+
+        # Split the output FIFOs for each column into groups corresponding to the cores
+        # in that column and link them to the memory tile FIFOs.
+        outC_fifos_split = [[], [], [], []]
+        for i in range(n_col):
+            outC_fifos_split[i] = outC_fifos[
+                i * n_cores_per_col : (i + 1) * n_cores_per_col
+            ]
+
+        # Link the output FIFOs from the cores to the memory tiles in each column
+        for i in range(n_col):
+            object_fifo_link(outC_fifos_split[i], of_out[i], of_c_offsets, [])
+
+        # Set up a packet-switched flow from core to shim for tracing information
+        tiles_to_trace = [cores[0]]
+        if trace_size > 0:
+            trace_utils.configure_packet_tracing_flow(tiles_to_trace, ShimTile)
+
+        # Set up compute tiles
+        for i in range(n_cores):
+            # Compute tile i
+            @core(cores[i], "kernels.a")
+            def core_body():
+                for _ in range_(0xFFFFFFFF):
+                    for _ in range_(tiles):
+                        elem_in = inA_fifos[i].acquire(ObjectFifoPort.Consume, 1)
+                        elem_out = outC_fifos[i].acquire(ObjectFifoPort.Produce, 1)
+
+                        softmax_bf16_vector(elem_in, elem_out, n)
+
+                        outC_fifos[i].release(ObjectFifoPort.Produce, 1)
+                        inA_fifos[i].release(ObjectFifoPort.Consume, 1)
+
+        # To/from AIE-array data movement
+        tensor_ty = np.ndarray[(N,), np.dtype[bfloat16]]
+
+        @runtime_sequence(tensor_ty, tensor_ty)
+        def sequence(A, C):
+
+            if trace_size > 0:
+                trace_utils.configure_packet_tracing_aie2(
+                    tiles_to_trace=tiles_to_trace,
+                    shim=ShimTile,
+                    trace_size=trace_size,
+                    trace_offset=N_in_bytes,
+                    ddr_id=1,
+                )
+
+            in_tasks = []
+            out_tasks = []
+
+            # Loop through each column to set up DMA tasks for input and output
+            for i in range(n_col):
+                # Distributing host buffer (A) to the shim tiles
+                in_tasks.append(
+                    shim_dma_single_bd_task(
+                        of_in[i],
+                        A,
+                        offset=N_per_shimtile * i,
+                        sizes=[1, 1, 1, N_per_shimtile],
+                        issue_token=True,
```

Comment:
Do we need to issue a token for the inputs and wait on them, or is it sufficient to do so for the outputs?